packages/plugin/src/rules/no-unreachable-types.ts

```diff
@@ -0,0 +1,80 @@
+import { GraphQLESLintRule } from '../types';
+import { requireReachableTypesFromContext } from '../utils';
+
+const UNREACHABLE_TYPE = 'UNREACHABLE_TYPE';
+
+const rule: GraphQLESLintRule = {
+  meta: {
+    messages: {
+      [UNREACHABLE_TYPE]: `Type "{{ typeName }}" in unreachable`,
```

Comment:
is* 
Do we need different message for other graphql kinds? like `{{ kind }} {{ name }} is unreachable` ? so it will be `scalar Email is unreachable by the loaded operations`.

Reply:
This rule is not about types being unused by operations, it's all within schema. Everything in schema starts with in fields of Query, Mutation or Subscription. The type is reachable if you can access it in those fields directly or indirectly any level deep. Take a look at the tests.

Reply:
There's another rule we can implement, something that I do in GraphQL Inspector which involves operations and fragments to detect unused pieces of schema.

Reply:
@dotansimha Scalar is a type so I think it's fine to use just `type`. We don't need more messages because we're tracking only type definitions and extensions, not their fields and values.